gateway/proxyproto/ssmproxy/ssm.go

```diff
@@ -0,0 +1,369 @@
+package ssmproxy
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/aws/session-manager-plugin/src/service"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+	"github.com/hoophq/hoop/common/grpc"
+	"github.com/hoophq/hoop/common/log"
+	pb "github.com/hoophq/hoop/common/proto"
+	pbagent "github.com/hoophq/hoop/common/proto/agent"
+	pbclient "github.com/hoophq/hoop/common/proto/client"
+	"github.com/hoophq/hoop/gateway/appconfig"
+	"github.com/hoophq/hoop/gateway/models"
+	"github.com/hoophq/hoop/gateway/proxyproto/grpckey"
+)
+
+var (
+	store       = sync.Map{}
+	instanceKey = "ssm_instance"
+	upgrader    = websocket.Upgrader{
+		ReadBufferSize:  1024,
+		WriteBufferSize: 1024,
+	}
+)
+
+type SSMProxy struct {
+	listener    net.Listener
+	listenAddr  string
+	router      gin.IRouter
+	connections atomic.Int32
+}
+
+// GetServerInstance returns the singleton instance of SSMServer.
+func GetServerInstance() *SSMProxy {
+	server, _ := store.LoadOrStore(instanceKey, &SSMProxy{})
+	return server.(*SSMProxy)
+}
+
+func (r *SSMProxy) AttachHandlers(router gin.IRouter) {
+	r.router = router
+	router.Handle(http.MethodGet, "/", func(c *gin.Context) {
+		c.String(http.StatusBadRequest, "Invalid request")
+	})
+	router.Handle(http.MethodPost, "/", r.handleStartSession)
+	router.Handle(http.MethodGet, "/ws/:connectionId", r.handleWebsocket)
```

Comment:
we kind already have the websocket https://github.com/hoophq/hoop/blob/main/gateway/api/server.go#L780 for rdp here which is not the sabe flow because we handle the hoop_key from the agentrs to validate the agent, but we kind have this entrypoint for websocket interface.

what would be the effort to add this r.handleWebsocket in the same transport, just asking because might not be the same flow of that to registry a agent